tables/Tables/TableUtil.cc

```diff
@@ -57,9 +57,16 @@ namespace casacore {
         return Table(tableName, lockOptions, option, tsmOption);
       }
       // Try to find and open the last subtable by splitting at ::
-      // It throws an exception if not possible.
-      std::pair<Table,String> t = tryFindSubTable (tableName, True);
-      return t.first;
+      std::pair<Table,String> t(findParentTable(tableName));
+      if (t.first.isNull()) {
+        throw TableError ("Table " + tableName + " does not exist");
+      }
+      // Find the last subtable in the parent.
+      if (! t.first.keywordSet().isDefined(t.second)) {
+        throw TableError ("Table name " + tableName + " is invalid (" +
+                          "subtable " + t.second + " is unknown)");
+      }
+      return t.first.keywordSet().asTable (t.second);
```

Comment:
You still don't use `lockOptions`, `option` and `tsmOption` when opening a subtable.

Reply:
Added them to findParentTable.